File: 217-contains-duplicate/contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        # s = set()
        # for n in nums:
        #     if n in s:
        #         return True
        #     s.add(n)
        # return False
        #return len(nums) != len(set(nums))
        dist = {}
        for i in range(len(nums)):
            if nums[i] in dist:
                return True
            dist[nums[i]] = i
        return False

print(Solution().containsDuplicate([1,2,3,1]))

File: 217-contains-duplicate/contains-duplicate.java
import java.util.*;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer, Integer> dist = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (dist.containsKey(nums[i])) {
                return true;
            }
            dist.put(nums[i], i);
        }
        return false;
    }
}

